### PR TITLE
add normalization module

### PR DESCRIPTION
**Pull request by jenna-tomkinson**
@gwaybio and @d33bs 

Please review this latest PR for CFReT. I added the normalization module to CFReT and edited code.

### CODE REVIEW

File: 2.cellprofiler_processing/README.md [hunk 1]
```diff
@@ -77,7 +77,7 @@ To run CellProfiler through the command line, run the [example_cp.ipynb notebook
 bash cellprofiler_feature_seg.sh
 ```
 
-CellProfiler runs on CPU, and it took approximately 9 hours and 50 minutes to run plate `localhost220512140003_KK22-05-198` and approximately 5 hours to run plate `localhost220513100001_KK22-05-198_FactinAdjusted`. 
+CellProfiler runs on CPU, and it took approximately 10 hours to run plate `localhost220512140003_KK22-05-198` and approximately 5 hours to run plate `localhost220513100001_KK22-05-198_FactinAdjusted`.
```
Comment (d33bs) [inline 2.cellprofiler_processing/README.md:80]: Noticing that this line changes every so often, it might be helpful to consider using a library or module to assist with this so it's automatically embedded (and referenceable here other readme's). Jupyter's [`%time`](https://ipython.readthedocs.io/en/stable/interactive/magics.html#magic-time) or using shell `$SECONDS` creatively (for example, see this [SO answer](https://stackoverflow.com/a/5153036)).

File: 3.processing_CP_features/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+### Step 2a: Set Up Paths
+
+Within the [extract_sc_features.ipynb](3.processing_CP_features/extract_sc_features.ipynb) notebook, you can change the paths to reflect the local paths or names for your machine (***IF* you changed anything from the original pipeline**) for the various parameters (e.g. CellProfiler directory, output directory, path to sqlite file, etc.)
+
+### Step 2b: Run Extract Single Cell Features
+
+Using the code below, run the notebook to extract and normalize single cell features from CellProfiler.
+
+```bash
```
Comment (d33bs) [inline 3.processing_CP_features/README.md:40]: Should this be changed to match the codeblock types above?

```suggestion
```sh
```

File: 3.processing_CP_features/extract_sc_features.ipynb [hunk 3]
```diff
@@ -0,0 +1,2827 @@
+{
```
Comment (d33bs) [inline 3.processing_CP_features/extract_sc_features.ipynb:1]: Should this notebook have a related `.py` nbconverted file checked in along with this PR?

File: 3.processing_CP_features/3.preprocessing_CFReT_features.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+name: 3.process-cfret-features
```
Comment (d33bs) [inline 3.processing_CP_features/3.preprocessing_CFReT_features.yml:1]: I noticed several variations of the names used for the environment, the folder, and these files. Would it make sense to unify or adjust the names to increase readability?

For reference:
- Environment name: `3.process-cfret-features`
- File name: `3.preprocessing_CFReT_features.yml`
- Folder name: `3.processing_CP_features`
- Notebook name: `extract_sc_features.ipynb` (uses `3.process-cfret-features` environment)

File: 2.cellprofiler_processing/cputils.py [hunk 5]
```diff
@@ -16,7 +16,9 @@ def rename_sqlite_file(sqlite_file_path: pathlib.Path, plate: str):
     """
     for sqlite_files in sqlite_file_path.iterdir():
         # the CellProfiler pipeline hardcodes the .sqlite file name, so all files outputted are called "CFReT.sqlite"
+        print(sqlite_files)
```
Comment (gwaybio) [inline 2.cellprofiler_processing/cputils.py:19]: generally we avoid having debugging print statements in production code unless it's important for logging.

Should we keep these print statements?

File: 3.processing_CP_features/README.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+```sh
+# Run this command in terminal to create the conda environment
+conda env create -f 3.preprocessing_CFReT_features.yml
+```
+
+## Step 2: Normalize Single Cell Features
+
+### Step 2a: Set Up Paths
+
+Within the [extract_sc_features.ipynb](3.processing_CP_features/extract_sc_features.ipynb) notebook, you can change the paths to reflect the local paths or names for your machine (***IF* you changed anything from the original pipeline**) for the various parameters (e.g. CellProfiler directory, output directory, path to sqlite file, etc.)
```
Comment (gwaybio) [inline 3.processing_CP_features/README.md:34]: sorry, I don't think I understand the bold sentence. Can you rephrase?

Comment (gwaybio) [reply 3.processing_CP_features/README.md:34]: there also may be something wonky going on with the asterisks.

File: 3.processing_CP_features/extract_sc_features.ipynb (hunk 3 above)
Comment (gwaybio) [inline 3.processing_CP_features/extract_sc_features.ipynb:1]: Can you create an `nbconverted` .py file?

Comment (gwaybio) [reply 3.processing_CP_features/extract_sc_features.ipynb:1]: (and add to a `3.processing_CP_features/scripts/nbconverted folder`)

**Review by gwaybio — APPROVED** [review]
Mostly minor comments. Overall, this looks great! 🤩 

I have one general question, and one general suggestion (both not necessary prior to merging).

## General question

Do you know why there are 2,013 features? I generally expect to see more - how many are in the original cell health data?

## General suggestion

In the future, you are likely going to need to process more than two plates a time. 

Your current approach, in `3.preprocessing_CFReT_features`, is to set new variables for each plate. It might be worth thinking about how this would scale poorly to, say, 25 plates. How would you approach this programmatically in this case?

**Review by d33bs — COMMENTED** [review]
Nice job with the changes in this PR! I left a few minor comments and suggestions. Please don't hesitate to let me know if you have any questions.

File: 3.processing_CP_features/3.preprocessing_CFReT_features.yml (hunk 4 above)
Comment (gwaybio) [reply 3.processing_CP_features/3.preprocessing_CFReT_features.yml:1]: this is a good catch @d33bs! Yes, I think it does make sense to unify naming!

File: 2.cellprofiler_processing/cputils.py (hunk 5 above)
Comment (jenna-tomkinson) [reply 2.cellprofiler_processing/cputils.py:19]: No that was my bad, I will remove it. I was having some troubles but didn't realize it was still there.

File: 3.processing_CP_features/README.md (hunk 6 above)
Comment (jenna-tomkinson) [reply 3.processing_CP_features/README.md:34]: I'll remove, I realized that there is no need for absolute paths so I will rephrase.

File: 3.processing_CP_features/extract_sc_features.ipynb (hunk 3 above)
Comment (jenna-tomkinson) [reply 3.processing_CP_features/extract_sc_features.ipynb:1]: :+1:

File: 2.cellprofiler_processing/README.md (hunk 1 above)
Comment (jenna-tomkinson) [reply 2.cellprofiler_processing/README.md:80]: Not sure how this would work, but definitely seems like something that can be implemented! I will discuss this with you!

File: 3.processing_CP_features/README.md (hunk 2 above)
Comment (jenna-tomkinson) [reply 3.processing_CP_features/README.md:40]: :+1:

File: 3.processing_CP_features/3.preprocessing_CFReT_features.yml (hunk 4 above)
Comment (jenna-tomkinson) [reply 3.processing_CP_features/3.preprocessing_CFReT_features.yml:1]: You are right, I didn't even notice! I will leave the notebook different since the it should be more descriptive imo. I have changed the rest. Thank you!

File: 3.processing_CP_features/extract_sc_features.ipynb (hunk 3 above)
Comment (jenna-tomkinson) [reply 3.processing_CP_features/extract_sc_features.ipynb:1]: Yup! I have just done it!

**Review by d33bs — APPROVED** [review]
